src/tools/qt-gui/src/treeviewmodel.cpp

```diff
@@ -540,73 +544,79 @@ void TreeViewModel::synchronize()
 		return;
 	}
 
-	ours = ours.cut(root);
-	theirs = theirs.cut(root);
-	base = base.cut(root);
-
-	ThreeWayMerge merger;
-
-	OneSideMergeConfiguration configuration(OURS);
-	configuration.configureMerger(merger);
-
-#if DEBUG && VERBOSE
-	theirs.rewind();
-	base.rewind();
-	for (Key o : ours)
-	{
-		Key t = theirs.next();
-		Key b = base.next();
-		std::cout << o.getName() << " " << ckdb::keyNeedSync(*o);
-		std::cout << "\t";
-		!b.isValid() ? std::cout << "none" : std::cout << b.getName() << " " << ckdb::keyNeedSync(*b);
-		std::cout << "\t";
-		!t.isValid() ? std::cout << "none" : std::cout << t.getName() << " " << ckdb::keyNeedSync(*t);
-		std::cout << std::endl;
-	}
-#endif
-
-	MergeResult result = merger.mergeKeySet(MergeTask(BaseMergeKeys(base, root),
-													  OurMergeKeys(ours, root),
-													  TheirMergeKeys (theirs, root),
-													  root));
-	if (!result.hasConflicts ())
+	try
 	{
-		resultKeys.append(result.getMergedKeys());
+		kdb.set(ours, root);
```

Comment:
Wasn't that the point of the whole thing? Set ours and in case of conflict merge the sets? Shouldn't that be the same as merging without conflicts?